domains/pallets/messenger/src/lib.rs

```diff
@@ -377,42 +378,26 @@ mod pallet {
         fn pre_dispatch(call: &Self::Call) -> Result<(), TransactionValidityError> {
             match call {
                 Call::relay_message { msg: xdm } => {
-                    let ValidatedRelayMessage {
-                        msg,
-                        next_nonce,
-                        should_init_channel,
-                    } = Self::validate_relay_message(xdm)?;
-                    if msg.nonce != next_nonce {
-                        log::error!(
-                            "Unexpected message nonce, channel next nonce {:?}, msg nonce {:?}",
-                            next_nonce,
-                            msg.nonce,
-                        );
-                        return Err(if msg.nonce < next_nonce {
-                            InvalidTransaction::Stale
-                        } else {
-                            InvalidTransaction::Future
-                        }
-                        .into());
-                    }
-                    Self::pre_dispatch_relay_message(msg, should_init_channel)
+                    let consensus_state_root = T::MmrProofVerifier::verify_proof_and_extract_leaf(
```

Comment:
I do not understand this refactor at all. It seems like you just moved the verification ahead instead. What am I missing ?
Would prefer some comments on why the refactor is necessary here

Reply:
It is used to extract the `T::MmrProofVerifier::verify_proof_and_extract_leaf` from the `validate_relay_message`